### PR TITLE
Support reusable action workflows and set up plugins' workflow

### PR DESCRIPTION
**Pull request by tianleh**
### Description

Support https://docs.github.com/en/actions/using-workflows/reusing-workflows and set up workflows for different plugins to avoid long running time on PR. Disable the current long running `.github/workflows/cypress-workflow-plugins-bundle-snapshot-based.yml`

`.github/workflows/release-e2e-workflow-template.yml` is the reusable workflow which handles the major functionalities and all other plugin workflows will call it by passing corresponding input parameters.

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianleh** [comment]
Can @opensearch-project/opensearch-dashboards-test help take a look?

File: .github/workflows/alerting-release-e2e-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    branches:
+      - main
+      - dev-*
+jobs:
+  tests:
+    uses: ./.github/workflows/release-e2e-workflow-template.yml
+    with:
+      test-name: Alerting
+      test-command: yarn cypress:run-with-security --browser chromium --spec 'cypress/integration/plugins/alerting-dashboards-plugin/*'
+      version: 1.3.0
```
Comment (kavilla) [inline .github/workflows/alerting-release-e2e-workflow.yml:17]: Do we need to change this every release?

File: .github/workflows/alerting-release-e2e-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Alerting Release tests workflow in Bundled OpenSearch Dashboards
+on:
+  pull_request:
+    branches:
+      - main
+      - dev-*
+  push:
```
Comment (kavilla) [inline .github/workflows/alerting-release-e2e-workflow.yml:7]: Do we on need on push?

**Review by kavilla — APPROVED** [review]
LGTM. Just some comments.

File: .github/workflows/release-e2e-workflow-template.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        type: string
+      version:
+        required: true
+        type: string
+env:
+  VERSION: ${{ inputs.version }}
+jobs:
+  tests:
+    name: Run Cypress E2E tests for ${{ inputs.test-name }}
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (tianleh) [reply .github/workflows/release-e2e-workflow-template.yml:19]: Created issue https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/141

File: .github/workflows/alerting-release-e2e-workflow.yml (hunk 1 above)
Comment (tianleh) [reply .github/workflows/alerting-release-e2e-workflow.yml:17]: Good suggestion. created issue to read from package.json https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/142

File: .github/workflows/alerting-release-e2e-workflow.yml (hunk 2 above)
Comment (tianleh) [reply .github/workflows/alerting-release-e2e-workflow.yml:7]: seems no too much value for running on push. Let me remove these.

**Pull request merged**